tests/integration/kubernetes/k8s-openvpn.bats

```diff
@@ -0,0 +1,86 @@
+    client_pod_template_yaml="${pod_config_dir}/pod-openvpn-client.yaml"
+    client_pod_instance_yaml="${pod_config_dir}/pod-openvpn-client-instance.yaml"
+
+    # Policy can be generated as to be populated Secrets are out of scope for policy
+    policy_settings_dir="$(create_tmp_policy_settings_dir "${pod_config_dir}")"
+    add_requests_to_policy_settings "${policy_settings_dir}" "ReadStreamRequest"
+    auto_generate_policy "${policy_settings_dir}" "$server_pod_template_yaml"
+    auto_generate_policy "${policy_settings_dir}" "$client_pod_template_yaml"
+}
+
+@test "Pods establishing a VPN connection using openvpn" {
+    # Step 1: Deploy the initialization pod and wait for it to be ready
+    kubectl apply -f "$init_pod_yaml"  && kubectl wait --for=condition=Ready --timeout=$timeout pod/$init_pod_name
+
+    # Step 2: Extract base64-encoded certificates from the initialization pod
+    export BASE64_CA_CRT="$(kubectl exec $init_pod_name -- cat /etc/openvpn/ca.crt.b64 | tr -d '\n')"
+    export BASE64_CLIENT_CRT="$(kubectl exec $init_pod_name -- cat /etc/openvpn/client.crt.b64 | tr -d '\n')"
+    export BASE64_CLIENT_KEY="$(kubectl exec $init_pod_name -- cat /etc/openvpn/client.key.b64 | tr -d '\n')"
+    export BASE64_SERVER_CRT="$(kubectl exec $init_pod_name -- cat /etc/openvpn/server.crt.b64 | tr -d '\n')"
+    export BASE64_SERVER_KEY="$(kubectl exec $init_pod_name -- cat /etc/openvpn/server.key.b64 | tr -d '\n')"
+    export BASE64_DH_PEM="$(kubectl exec $init_pod_name -- cat /etc/openvpn/dh.pem.b64 | tr -d '\n')"
+
+    [ -n "$BASE64_CA_CRT" ]
+    [ -n "$BASE64_CLIENT_CRT" ]
+    [ -n "$BASE64_CLIENT_KEY" ]
+    [ -n "$BASE64_SERVER_CRT" ]
+    [ -n "$BASE64_SERVER_KEY" ]
+    [ -n "$BASE64_DH_PEM" ]
+
+    # Step 3: Substitute environment variables in template files, write to instance files
+    envsubst < "$server_pod_template_yaml" > "$server_pod_instance_yaml"
+    envsubst < "$client_pod_template_yaml" > "$client_pod_instance_yaml"
+
+    # Step 4: Deploy the OpenVPN server and wait for it to be ready (uses readiness probe)
+    kubectl apply -f "$server_pod_instance_yaml" && kubectl wait --for=condition=Ready --timeout=$timeout pod/$server_pod_name
+
+    # Step 5: Deploy the OpenVPN client and wait for it to be ready (uses readiness probe)
+    kubectl apply -f "$client_pod_instance_yaml" && kubectl wait --for=condition=Ready --timeout=$timeout pod/$client_pod_name
+}
+
+teardown() {
+    # Debugging information
+    echo "=== OpenVPN Init Pod Logs ==="
+    kubectl logs "$init_pod_name" --all-containers=true || true
+    echo "=== OpenVPN Server Pod Logs ==="
+    kubectl logs "$server_pod_name" || true
+    echo "=== OpenVPN Client Pod Logs ==="
+    kubectl logs "$client_pod_name" || true
+
+    # Describe pods for additional debugging
+    kubectl describe pod "$init_pod_name" || true
+    kubectl describe pod "$server_pod_name" || true
+    kubectl describe pod "$client_pod_name" || true
+
```

Comment:
teardown_common() describes all pods, can remove.

Reply:
Curious where you see this. When I comment these lines, I see no `kubectl describe` output.